wallet-new/src/Cardano/Wallet/API/V1/Migration/Types.hs

```diff
@@ -121,6 +122,11 @@ instance Migrate V0.CCoin (V1 Core.Coin) where
 instance Migrate (V1 Core.Coin) V0.CCoin where
     eitherMigrate (V1 c) = pure (V0.encodeCType c)
 
+instance (n ~ m, ValidMnemonicSentence n)
+    => Migrate (Mnemonic n) (V0.CBackupPhrase m) where
+    eitherMigrate =
+        Right . V0.CBackupPhrase
```

Comment:
HOP HOP HOP! Not so fast hahaha :joy: 
Redemption backup phrase are 9-word mnemonics and are distinct from the wallet backup phrases (12-word mnemonics). Even though both turn into a `CBackupPhrase` for backward compatibility, I wonder if it wouldn't be worth it to make this explicit and have two separate instances for `Mnemonic 9` and `Mnemonic 12`. 

Especially when we talked about possibly longer mnemonics for new features that won't be turned into a `CBackupPhrase` at all.

Reply:
Great catch! I'll create a type family that errors out if it's not 9 or 12 :)